cave/plot/plotter.py

```diff
@@ -337,3 +336,71 @@ def plot_cost_over_time(self, rh, traj, output="performance_over_time.png",
         plt.tight_layout()
         fig.savefig(output)
         plt.close(fig)
+
+    def plot_cost_over_time(self, rh, traj, output="performance_over_time.png",
+                            validator=None):
+        """ Plot performance over time, using all trajectory entries
+            with max_time = wallclock_limit or (if inf) the highest
+            recorded time
+
+            Parameters
+            ----------
+            rh: RunHistory
+                runhistory to use
+            traj: List
+                trajectory to take times/incumbents from
+            output: str
+                path to output-png
+            epm: RandomForestWithInstances
+                emperical performance model (expecting trained on all runs)
+        """
+        self.logger.debug("Estimating costs over time for best run.")
+        validator.traj = traj  # set trajectory
+        time, configs = [], []
+
+        mean, var = [], []
+        c = []
+        for entry in traj:
+            time.append(entry["wallclock_time"])
+            configs.append(entry["incumbent"])
+            costs = _cost(configs[-1], rh, rh.get_runs_for_config(configs[-1]))
+            print(len(costs), time[-1])
+            if not costs:
+                time.pop()
```

Comment:
So I understand this is a sketch. This is the point where we'd have to decide what to do with configs not evaluated on all instances? Currently, this only removes configs/timesteps which where evaluated on no instances, right? I think we can do this, but we should definitely estimate costs for missing instances, otherwise one single instance would falsely result in a  low uncertainty, or am I mislead?

Reply:
Yes it is and I agree.
I think for the cost over time plot we could use the validator to fill the missing information in.
Do you agree?

Reply:
We could. How do we propagate the uncertainty of the epm? Or do we treat the missing information the same way as actual evaluations?